cmd/minishift/cmd/artifact_preflight.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	configCmd "github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/minishift/minishift/pkg/minishift/oc"
+	"github.com/minishift/minishift/pkg/util"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// preflightChecksForArtifacts is executed once artifacts are cached.
+func preflightChecksForArtifacts() {
+	artifactsCheckSucceedsOrFails(checkOcFlag, "Checking if provided oc flags are supported", "Provided oc flag not supported")
+}
+
+// artifactCheckFunc returns true when check passed
+type artifactCheckFunc func() bool
+
+// artifactsCheckSucceedsOrFails executes a pre-flight test function and prints
+// the returned status in a standardized way. If the test fails and returns a
+// false, the application will exit with errorMessage to describe what the
+// cause is.
+func artifactsCheckSucceedsOrFails(execute artifactCheckFunc, message string, errorMessage string) {
+	fmt.Printf("-- %s ... ", message)
+
+	if execute() {
+		fmt.Println("OK")
+		return
+	}
+	fmt.Println("FAIL")
+	atexit.ExitWithMessage(1, errorMessage)
+}
+
+// checkOcFlag checks if provided oc flags are supported
+func checkOcFlag() bool {
+	clusterUpParams := determineInitialClusterupParameters()
+	for _, key := range clusterUpParams {
+		if !oc.SupportFlag(key, ocPath, &util.RealRunner{}) {
```

Comment:
Took me some time to figure out from where this `ocPath` is coming from. It is being set in `start.go`. Not sure how to mitigate it except refactoring. 
One possible way is to have another folder which include all pre-flight checks files.

Reply:
Yes, that going to be a big refactor because right now we are leveraging a lot of unexported variable in different part of the preflight check if we move then we have restructured around.

Reply:
`ocPath` is used as a global static. this is confusing and should be fixed in follow-ups

Reply:
> One possible way is to have another folder which include all pre-flight checks files.

+1.  We should fix this in a flow up issue.